class_allocation.py

```diff
@@ -299,78 +301,66 @@ def ecpc(self):
         Equal Cumulative per Capita: Uses historical emissions, discount factors and
         population shares to allocate the global budget
         '''
-        # make sqrt curve
-        compensation_form_sqrt = np.sqrt(np.arange(0, 2101 - self.start_year_analysis))
-        # sum of values has to be 1
-        compensation_form_sqrt = compensation_form_sqrt / np.sum(compensation_form_sqrt)
-
-        xr_comp = xr.DataArray(compensation_form_sqrt, dims=['Time'], coords={'Time': self.analysis_timeframe})
-
         # Defining the timeframes for historical and future emissions
-        xrs = []
-        hist_emissions_startyears = self.settings['params']['hist_emissions_startyears']
-        discount_rates = self.settings['params']['discount_rates']
+        population_data = self.xr_total.Population.sel(Time=self.analysis_timeframe)
+        global_emissions_future = self.xr_total.GHG_globe.sel(Time=self.analysis_timeframe)
+        GF_frac = self.xr_total.GHG_hist.sel(Time=2021, Region=self.focus_region) / self.xr_total.GHG_hist.sel(Time=2021, Region='EARTH')
```

Comment:
Better to use `self.start_year_analysis` here instead of hardcoding the year 2021.